plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1936,6 +1936,76 @@ public void testCreateNestedPartitionedTable()
         dropTable("test_nested_table_3");
     }
 
+    @Test
+    public void testCreateWithNestedPartitionedColTable()
+    {
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE test_nested_partition_table_1 (" +
+                " bool BOOLEAN" +
+                ", int INTEGER" +
+                ", arr ARRAY(VARCHAR)" +
+                ", big BIGINT" +
+                ", rl REAL" +
+                ", dbl DOUBLE" +
+                ", mp MAP(INTEGER, VARCHAR)" +
+                ", dec DECIMAL(5,2)" +
+                ", vc VARCHAR" +
+                ", vb VARBINARY" +
+                ", ts TIMESTAMP(6)" +
+                ", str ROW(id INTEGER , vc VARCHAR)" +
+                ", dt DATE)" +
+                " WITH (partitioning = ARRAY['str.id'])";
+
+        assertUpdate(createTable);
+
+        @Language("SQL") String insertSql = "INSERT INTO test_nested_partition_table_1 " +
+                " select true, 1, array['uno', 'dos', 'tres'], BIGINT '1', REAL '1.0', DOUBLE '1.0', map(array[1,2,3,4], array['ek','don','teen','char'])," +
+                " CAST(1.0 as DECIMAL(5,2))," +
+                " 'one', VARBINARY 'binary0/1values',\n" +
+                " cast(current_timestamp as TIMESTAMP), (CAST(ROW(1, 'this is a random value') AS ROW(int, varchar))), current_date";
+
+        assertUpdate(insertSql, 1);
+        MaterializedResult result = computeActual("SELECT * from test_nested_partition_table_1");
+        assertEquals(result.getRowCount(), 1);
+
+        dropTable("test_nested_partition_table_1");
+
+        @Language("SQL") String createTable2 = "" +
+                "CREATE TABLE test_nested_partition_table_2 (" +
+                " int INTEGER" +
+                ", arr ARRAY(ROW(id INTEGER, vc VARCHAR))" +
+                ", big BIGINT" +
+                ", rl REAL" +
+                ", dbl DOUBLE" +
+                ", mp MAP(INTEGER, ARRAY(VARCHAR))" +
+                ", dec DECIMAL(5,2)" +
+                ", str ROW(id INTEGER, vc VARCHAR, arr ARRAY(INTEGER))" +
+                ", vc VARCHAR)" +
+                " WITH (partitioning = ARRAY['str.id'])";
+
+        assertUpdate(createTable2);
```

Comment:
how is second table different from the first?
ie what is the important difference?